pkg/sql/mvcc_backfiller.go

```diff
@@ -70,6 +72,10 @@ func (sc *SchemaChanger) Merge(
 
 			wb := txn.NewBatch()
 			for i := range kvs {
+				if kvs[i].Value.Timestamp.Less(readAsOf) {
```

Comment:
I believe it is required for correctness. I've explained the case it is handling in the commit message, but can also put a comment here.

Reply:
Ah. Okay, caught up on the commit message and now see why it's here. Let me ponder a bit, since I'm not sure this will work on top of restored data where we forward all timestamps.

Reply:
Yeah.  Another option (echoing from slack) is to persist the timestamp in the body of a preserved delete so that it could survive the restore.

Reply:
This has been made unnecessary by not issuing delete-preserving deletes when the index is in DeleteOnly. See #75720